|                |   |
:----------------|---|
| **Nombre**     | Jocelyn Jiménez Buenrostro
| **Fecha**      | 09/02/26  |
| **Expediente** |  738498 |

In [ ]:
import pandas as pd
from google.colab import files

df = pd.read_excel("Elfeo (4).xlsx")
df.head()

Tamaño Hielo     Sabor  Score
0  Grande   Sin    Kahlúa    8.2
1  Grande   Con    Kahlúa    9.0
2   Chica   Sin    Kahlúa    7.0
3   Chica   Con    Kahlúa    7.5
4  Grande   Con  Horchata   10.0

Ahora la tabla la transformamos en 0 y 1

grande= 1
Chico = 0

Sin = 0
con = 1

In [ ]:
df['Tamaño']= df['Tamaño'].replace({"Grande":1 , "Chica":0})
df['Hielo']= df['Hielo'].replace({"Con":1 , "Sin":0})

/tmp/ipython-input-4209904381.py:1: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['Tamaño']= df['Tamaño'].replace({"Grande":1 , "Chica":0})
/tmp/ipython-input-4209904381.py:2: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['Hielo']= df['Hielo'].replace({"Con":1 , "Sin":0})


In [ ]:
df["Kahlúa"] = (df["Sabor"] == "Kahlúa").astype(int)
df["Horchata"] = (df["Sabor"] == "Horchata").astype(int)

In [ ]:
df

Tamaño  Hielo     Sabor  Score  Kahlúa  Horchata
0       1      0    Kahlúa    8.2       1         0
1       1      1    Kahlúa    9.0       1         0
2       0      0    Kahlúa    7.0       1         0
3       0      1    Kahlúa    7.5       1         0
4       1      1  Horchata   10.0       0         1
5       0      0  Horchata    6.0       0         1
6       0      1  Horchata    6.4       0         1
7       1      0  Horchata    9.5       0         1

# Parte 1

**Regresión Lineal**

In [ ]:
import statsmodels.api as sm

In [ ]:
xf = df[["Tamaño","Hielo","Kahlúa","Horchata"]]
xf1 = sm.add_constant(xf)

In [ ]:
y = df["Score"].values.reshape(-1,1)

In [ ]:
xff = sm.add_constant(xf1)
model = sm.OLS(y,xff)
results = model.fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Mon, 09 Feb 2026   Prob (F-statistic):             0.0473
Time:                        23:38:23   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.3000      0.320     13.417      0.000       3.410       5.190
Tamaño         2.4500      0.555      4.414      0.012       0.909       3.991
Hielo          0.5500      0.555      0.991      0.378      -0.991       2.091
Kahlúa         2.1250      0.320      6.631      0.003       1.235       3.015
Horchata       2.1750      0.320      6.787      0.002       1.285       3.065
==============================================================================
Omnibus:                        7.172   Durbin-Watson:                   1.431
Prob(Omnibus):                  0.028   Jarque-Bera (JB):                1.242
Skew:                          -0.048   Prob(JB):                        0.537
Kurtosis:                       1.072   Cond. No.                     2.22e+16
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 3.37e-32. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

# Parte 2

sin = -1
con = 1

Grande -1
chica 1

In [ ]:
df_1 = pd.read_excel("Elfeo (4).xlsx")
df_1.head()

Tamaño Hielo     Sabor  Score
0  Grande   Sin    Kahlúa    8.2
1  Grande   Con    Kahlúa    9.0
2   Chica   Sin    Kahlúa    7.0
3   Chica   Con    Kahlúa    7.5
4  Grande   Con  Horchata   10.0

In [ ]:
df_1['Tamaño']= df_1['Tamaño'].replace({"Grande":-1 , "Chica":1})
df_1['Hielo']= df_1['Hielo'].replace({"Con":1 , "Sin":-1})

/tmp/ipython-input-3473666795.py:1: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df_1['Tamaño']= df_1['Tamaño'].replace({"Grande":-1 , "Chica":1})
/tmp/ipython-input-3473666795.py:2: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df_1['Hielo']= df_1['Hielo'].replace({"Con":1 , "Sin":-1})


In [ ]:
df_1["Kahlúa"] = (df_1["Sabor"] == "Kahlúa").astype(int)
df_1["Horchata"] = (df_1["Sabor"] == "Horchata").astype(int)

In [ ]:
df_1

Tamaño  Hielo     Sabor  Score  Kahlúa  Horchata
0      -1     -1    Kahlúa    8.2       1         0
1      -1      1    Kahlúa    9.0       1         0
2       1     -1    Kahlúa    7.0       1         0
3       1      1    Kahlúa    7.5       1         0
4      -1      1  Horchata   10.0       0         1
5       1     -1  Horchata    6.0       0         1
6       1      1  Horchata    6.4       0         1
7      -1     -1  Horchata    9.5       0         1

**Regresión lineal**

In [ ]:
xf2 = df_1[["Tamaño","Hielo","Kahlúa","Horchata"]]
xf12 = sm.add_constant(xf2)

In [ ]:
y1 = df_1["Score"].values.reshape(-1,1)

In [ ]:
xff2 = sm.add_constant(xf12)
model2 = sm.OLS(y1,xff2)
results2 = model2.fit()
results2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Mon, 09 Feb 2026   Prob (F-statistic):             0.0473
Time:                        23:52:18   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          5.3000      0.185     28.644      0.000       4.786       5.814
Tamaño        -1.2250      0.278     -4.414      0.012      -1.996      -0.454
Hielo          0.2750      0.278      0.991      0.378      -0.496       1.046
Kahlúa         2.6250      0.293      8.973      0.001       1.813       3.437
Horchata       2.6750      0.293      9.143      0.001       1.863       3.487
==============================================================================
Omnibus:                        7.172   Durbin-Watson:                   1.431
Prob(Omnibus):                  0.028   Jarque-Bera (JB):                1.242
Skew:                          -0.048   Prob(JB):                        0.537
Kurtosis:                       1.072   Cond. No.                     1.84e+16
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 3.54e-32. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

**Conclusión**

Como se puede observar en ambas regresiones se encuntra en R2 del 83.7%, por lo tanto, no indica que las variables incluidas en el modelo explican el score. Al cambiar (0,1) a (-1,1), modifica los coeficientes del modelo pero esto mismo no modifica como tal el modelo. Al final de cuentas es el mismo modelo, pero con diferentes valores.